### PR TITLE
.NET Core debugging with LCOW

### PR DESCRIPTION
**Pull request by philliphoff**
Add support for .NET Core debugging when using Docker for Windows with LCOW (Linux Containers On Windows) enabled.  Enabling LCOW allows Linux and Windows containers to be used at the same time. 
 (Without LCOW, only Linux *or* Windows containers can be used at any given time.) However, when this feature is enabled, Docker assumes Windows containers are built/run by default.  To build/run Linux containers, the `--platform Linux` argument must be specified.

### CODE REVIEW

File: debugging/coreclr/dockerClient.ts [hunk 1]
```diff
@@ -8,10 +8,15 @@ import { ProcessProvider } from "./processProvider";
 export type DockerBuildImageOptions = {
     context?: string;
     dockerfile?: string;
+    platform?: string;
```
Comment (StephenWeatherford) [inline debugging/coreclr/dockerClient.ts]: Should this be strongly typed to specific values?

File: debugging/coreclr/dockerClient.ts [hunk 2]
```diff
@@ -34,6 +39,7 @@ export type DockerRunContainerOptions = {
     command?: string;
     containerName?: string;
     entrypoint?: string;
+    platform?: string;
```
Comment (StephenWeatherford) [inline debugging/coreclr/dockerClient.ts]: ditto

File: debugging/coreclr/prereqManager.ts [hunk 3]
```diff
@@ -21,17 +21,28 @@ export type ShowErrorMessageFunction = (message: string, ...items: vscode.Messag
...
     public async checkPrerequisite(): Promise<boolean> {
         const daemonOsJson = await this.dockerClient.getVersion({ format: '{{json .Server.Os}}' });
-        const daemonOs = JSON.parse(daemonOsJson.trim());
+        const daemonOs = <string>JSON.parse(daemonOsJson.trim());
 
         if (daemonOs === 'linux') {
             return true;
         }
 
+        if (this.osProvider.os === 'Windows') {
```
Comment (StephenWeatherford) [inline debugging/coreclr/prereqManager.ts]: duplicate code

**Comment by EricJizbaMSFT** [comment]
FYI this PR was created before our recent switch from Travis CI to Azure Dev Ops. Push a commit (merging with master or addressing comments) and the "checks" for this PR will be updated

**Comment by EricJizbaMSFT** [comment]
Closing this PR due to lack of activity. Feel free to re-open after fixing merge conflicts

**Pull request closed without merging**